### PR TITLE
Add description for different types of blocks

### PR DESCRIPTION
**Pull request by MAX-786**
#3827 

Added description of Volto Blocks and for every different type of block. Also added relevant screenshots.

### CODE REVIEW

**Review by tiberiuichim — APPROVED** [review]

File: docs/source/user-manual/blocks.md [hunk 1]
```diff
@@ -54,32 +56,193 @@ Now with your empty block available, you can select its type in one of two ways.
 
 ## Description Block
 
-A description block provides a blank textarea to write plain text.
+These blocks allow users to add text content to a page. They can be configured to display different fonts, sizes, and colors, and they can also include links and formatting options like bold and italic.
```
Comment (ksuess) [inline docs/source/user-manual/blocks.md:59]: No, the description block is plain text and is in sync with the field "description", which should be plain text for SEO.

File: docs/source/user-manual/blocks.md [hunk 2]
```diff
@@ -54,32 +56,193 @@ Now with your empty block available, you can select its type in one of two ways.
...
+
 (user-manual-hero-block-label)=
 
 ## Hero Block
 
+A hero block is a type of block in Volto that is used to create a full-width banner or header for a page.
+It is typically used to highlight important content or to create a visual impact at the top of a page.
+
+Hero blocks typically include a background image or color, and they can also include text.
+To use a hero block in Volto, you can add it to your page by by choosing from the list of Blocks.
```
Comment (ksuess) [inline docs/source/user-manual/blocks.md:102]:
```suggestion
To use a hero block in Volto, you can add it to your page by choosing from the list of blocks.
```

**Review by ksuess — CHANGES_REQUESTED** [review]
This is definitely an enhancement. 
Would you mind adding also a section for the search block?

File: docs/source/user-manual/blocks.md (hunk 1 above)
Comment (MAX-786) [reply docs/source/user-manual/blocks.md:59]: should I replace it wtih this -> `The description block is a simple block that can be used to add plain text` or the previous description works -> `A description block provides a blank textarea to write plain text.` Thanks!

Comment (stevepiercy) [reply docs/source/user-manual/blocks.md:59]:
```suggestion
A description block accepts unformatted plain text.
When displayed, it appears as the description in the page, and for search engine optimization in HTML meta tags as `<meta name="description" content="DESCRIPTION_TEXT" data-react-helmet="true">` and `<meta property="og:description" content="DESCRIPTION_TEXT" data-react-helmet="true">`.
```

**Review by ksuess — APPROVED** [review]

**Review by stevepiercy — APPROVED** [review]
Thank you for starting this page. Animated gifs are awesome! I started a review, but it got too long and complicated with lots of nitpicky details. Yet you inspired me to take your work further by making many enhancements. Perhaps too many :grin:.

Here's a summary of the nitpicks, and general guidance to writing documentation—some of which I realized only while working on this page (and which I should add to our Docs Contributing Guide):

- Use dashes `-` in filenames and avoid underscores.
- Images should be no wider than 740 pixels to fit within the documentation's main view port. This avoids scaling and reducing legibility. To make that work in Volto, set your browser width to 1180 pixels. You will notice that the drag and trash icons for each block move inside the block from outside. This is a TIL (thing I learned).
- Screenshots of each step are helpful. It is painstaking, but totally worth it.
- Provide sufficient detail of what each option is and does.
- Avoid repetitive statements.
- Avoid marketing speak. Words such as "powerful", "easy", "simple" come off as hyping a feature, instead of plainly describing what it is and does.
- Avoid empty phrases and clauses. Short, terse sentences are preferred. For example:
  - "An images grid block is a type of block in Volto that allows users to display a grid of images on a page."
  - "An images grid block displays a row of images on a page."
- Headings should be Sentence cased, not Title Cased.

**Review by sneridagh — CHANGES_REQUESTED** [review]
@MAX-786 the grid and the system ones are not official Plone/Volto blocks, although they are present in the demo.

I'd leave the grid one for now (it's planned for Plone 6.1) and remove the system one, since it's a demo site block only.

For the rest, great addition!

**Comment by MAX-786** [comment]
Thank you so much for your thorough review and for taking the time to provide such thoughtful feedback. Your insights and suggestions are greatly appreciated and will be very valuable to me.

> - Provide sufficient detail of what each option is and does.

I thought of doing it but it made me nervous as it got too long 😖, I'll make sure nxt time.

> - Avoid repetitive statements.
> - Avoid marketing speak. Words such as "powerful", "easy", "simple" come off as hyping a feature, instead of plainly describing what it is and does.

I am wondering if it would be better to apply these changes now or wait until a later time?
Thanks, everyone for your help.

**Comment by MAX-786** [comment]
> @MAX-786 the grid and the system ones are not official Plone/Volto blocks, although they are present in the demo.
> 
> I'd leave the grid one for now (it's planned for Plone 6.1) and remove the system one, since it's a demo site block only.
> 
> For the rest, great addition!

Thanks for the review, Can you please tell me if these changes should be done by me or if you have already done them, as I can still see the System Block in the last commit?

**Comment by stevepiercy** [comment]
@MAX-786 please make the requested changes from @sneridagh. Thank you!

**Comment by MAX-786** [comment]
It's Done! Thanks.

**Comment by tiberiuichim** [comment]
OMG you're actually doing this! Writing a user manual for Volto. You two are awesome!

**Comment by MAX-786** [comment]
And as I have limited knowledge of cypress..... this 1 failing test is giving me horror, something's wrong in `listingblock.js` 
Below is the error,  I'll try googling if i can find what's the problem ..... and.....please help (>_<)
> Timed out retrying after 4050ms: cy.click() failed because this element is detached from the DOM.
> 
> `<button class="ui basic icon button listing">...</button>`
> 
> Cypress requires elements be attached in the DOM to interact with them.
> 
> The previous command that ran was:
> 
>   > cy.contains()
> 
> This DOM element likely became detached somewhere between the previous and current command.
> 
> Common situations why this happens:
> 
> - Your JS framework re-rendered asynchronously
>- Your app code reacted to an event firing and removed the element
> - You typically need to re-query for the element or add 'guards' which delay Cypress from running new commands.

**Comment by stevepiercy** [comment]
I think you can ignore the failing test as this PR involves documentation only, and nothing to do with core.

Nonetheless, how does one rerun a failed test in Cypress?

> OMG you're actually doing this! Writing a user manual for Volto. You two are awesome!

It ain't gonna write itself. 😝

**Comment by stevepiercy** [comment]
nm, I got lost in Cypress. I found it in GHA. Rerunning it now.

**Comment by stevepiercy** [comment]
The failing test was a glitch, and we're all green. Merging! 🥳

**Pull request merged**